### ISSUE TITLE
Fix S3440 FN: Add support for 'is' and 'is not' expressions

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6447]
### Description

The rule S3440 is not raising an issue for 'is' and 'is not' expressions.

### Repro steps

```csharp
          string x = null;

          if (!(x is null)) // FN (is expression not supported yet)
          {
              x = null;
          }

          if (x is not null) // FN (is not expression not supported yet)
          {
              x = null;
          }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

**Comment by mary-georgiou-sonarsource** [comment]
Please take into account that assigning big structure or `stackalloc` arrays or `span` can have a considerable cost which means that these checks might have an actual value.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1798](https://sonarsource.atlassian.net/browse/NET-1798)

[NET-1798]: https://sonarsource.atlassian.net/browse/NET-1798?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
